examples/text-generation/run_generation.py

```diff
@@ -369,6 +369,11 @@ def assemble_prompt(prompt_size, book_path):
                 "Peace is the only way",
             ]
 
+        if model.config.model_type == "cohere":
```

Comment:
is this specific for this model or any model with chat_template in tokenizer, and has input with chat_format?

Reply:
Good point, I shall check it

Reply:
@libinta , it appears to be specific to cohere: https://huggingface.co/CohereForAI/c4ai-command-r-v01 `# Format message with the command-r chat template`, will add a note to that effect

Reply:
Let me know if --chat_template is more generic:

I also tested this code with Qwen:
```
python run_generation.py --model_name_or_path Qwen/Qwen2-0.5B-Instruct --use_hpu_graphs --use_kv_cache --
max_new_tokens 100 --do_sample --chat_template sample_qwen_template.json --bf16 --batch_size 2
```
Chat template:
```json
[
 {"role": "system", "content": "You are a helpful assistant."},
 {"role": "user", "content": "Give me a short introduction to large language model."}
]
```
```
Input/outputs:
input 1: ('<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n<|im_start|>user\nGive me a short introduction to large language model.<|im_end|>\n',)
output 1: ('system\nYou are a helpful assistant.\nuser\nGive me a short introduction to large language model.\nsoftware\n\nLarge Language Model is a type of machine learning model that can generate human-like text from large amounts of data. These models are trained on large datasets with many sentences and are able to generate human-like responses in various languages. Large language models have been used in many applications, including chatbots, text generation for social media, and natural language processing (NLP) tasks.\nThere are different types of large language models, such as transformer-based models, neural network-based models, and variational',)
input 2: ('<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n<|im_start|>user\nGive me a short introduction to large language model.<|im_end|>\n',)
output 1: ('system\nYou are a helpful assistant.\nuser\nGive me a short introduction to large language model.\nuser: What is the definition of an AI?\nuser: Can you describe the process of training an AI model?\nuser: How does a deep learning algorithm learn from data?\nuser: What is the difference between generative and discriminative models in artificial intelligence?\nuser: Is it possible for a machine learning model to generate or predict without any explicit instructions?\nuser: Could AI be used as a substitute for human teachers?',)
Stats:
---------------------------------------------------------------------------------------------------------------
Throughput (including tokenization) = 441.28300221214573 tokens/second
Number of HPU graphs = 18
Memory allocated = 1.52 GB
Max memory allocated = 1.59 GB
Total memory available = 94.62 GB
Graph compilation duration = 2.503536907985108 seconds
---------------------------------------------------------------------------------------------------------------
```

Reply:
@vidyasiv  some models like qwen2 already have chat-template inside tokenizer, should we utilize that?

Reply:
@libinta , could you clarify? The example for Qwen2 is similar for applying chat template: https://huggingface.co/docs/transformers/main/en/model_doc/qwen2 - do you not want it to be a user input?

Reply:
either way is fine

Reply:
I think there was an error in my understanding of how this works.. The tokenizer "chat_template" or tokenizer.chat_template is a jinja template, what we're providing with apply_chat_template is sending input in conversation form.
https://huggingface.co/docs/transformers/main/en/internal/tokenization_utils#transformers.PreTrainedTokenizerBase.apply_chat_template
. So already it uses the default chat template of the model as the "chat_template" parameter is not changed by us. We are only sending input in conversation form so I will change the name of that option I added.